integration.go

```diff
@@ -32,28 +47,52 @@ type IntegrationMessageHandler interface {
 // In the context of this interface, "the handler" refers to the handler on
 // which Configure() has been called.
 type IntegrationConfigurer interface {
-	// Name sets the name of the handler. Each handler within an application must
-	// have a unique name.
+	// Name sets the name of the handler.
+	//
+	// It MUST be called exactly once within a single call to Configure().
+	//
+	// Each handler within an application MUST have a unique, non-empty name.
 	Name(n string)
 
-	// AcceptsCommandType configures the engine to route command messages of the
-	// same type as m to the handler.
-	AcceptsCommandType(m Message)
+	// ConsumesCommandType configures the engine to route command messages of
+	// the same type as m to the handler.
+	//
+	// It MUST be called at least once within a call to Configure(). It MUST NOT
+	// be called more than once with a command message of the same type.
+	//
+	// A given command type MUST be routed to exactly one handler within an
+	// application.
+	//
+	// The "content" of m MUST NOT be used, inspected, or treated as meaningful
+	// in any way, only its runtime type information.
```

Comment:
This last part feels a bit off.
> only its runtime type information.

Maybe end the sentence and reference "m" again? Something like:
> in any way. Only the runtime type information of "m" should be used.